h/api_client/subscribers.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+import pyramid.events
+
+import h.api_client.api_client as api_client
+
+
+def _get_api_url(request):
+    """Return the configured base URL for this app's Hypothesis API instance.
+
+    Returns the value of the h.api_url setting from the config file,
+    or if that doesn't exist the "/api" route from the WSGI app (which assumes
+    that we're running the app with the "api" feature enabled).
+
+    Always returns the URL _without_ a trailing /, despite what the user might
+    have entered in the config file.
+
+    """
+    return request.registry.settings.get(
+        "h.api_url", request.resource_url(request.root, "api")).rstrip("/")
```

Comment:
I was going to say exactly the same thing. :-)
On Mar 19, 2015 8:33 AM, "Nick Stenning" <notifications@github.com> wrote:

> In h/api_client/subscribers.py
> <https://github.com/hypothesis/h/pull/2029#discussion_r26749145>:
>
> > +import h.api_client.api_client as api_client
> > +
> > +
> > +def _get_api_url(request):
> > +    """Return the configured base URL for this app's Hypothesis API instance.
> > +
> > +    Returns the value of the h.api_url setting from the config file,
> > +    or if that doesn't exist the "/api" route from the WSGI app (which assumes
> > +    that we're running the app with the "api" feature enabled).
> > +
> > +    Always returns the URL _without_ a trailing /, despite what the user might
> > +    have entered in the config file.
> > +
> > +    """
> > +    return request.registry.settings.get(
> > +        "h.api_url", request.resource_url(request.root, "api")).rstrip("/")
>
>  Yeah, so there's already request.api_client.base_url actually, maybe
> just rename that to api_url
>
> No, just leave it as is. You don't need to rename it. It's the base_url --
> the fact it belongs to the API is implicit from the context.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hypothesis/h/pull/2029/files#r26749145>.
>